src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,268 @@
 package co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
 
-public class Range<T> implements Set<T> {
+public class Range<T extends Comparable<T>> implements Set<T> {
+
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<V> {
+        V item;
+        Node<V> next;
+        Node<V> prev;
+
+        public Node(V item) {
+            this.item = item;
+        }
+
+        Node(Node<V> prev, V element, Node<V> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    public static Range<Integer> of(Integer start, Integer end) {
+        Range<Integer> range = new Range<>();
+        if (start < end) {
+            for (Integer i = start; i <= end; i++) {
+                range.add(i);
+            }
+        }
+        return range;
+    }
+
+    public static Range<Float> of(Float start, Float end) {
```

Comment:
methods "of" are quite similar. Try to avoid using duplication of code (call of(start, end, func)